proxy.go

```diff
@@ -28,38 +29,50 @@ func copyHeader(dst, src http.Header) {
 }
 
 type Proxy struct {
-	config        *ProxyConfig
-	httpClient    *http.Client
-	remoteURL     *url.URL
-	signer        Signer
-	authenticator Authenticator
+	config                 *ProxyConfig
+	httpClient             *http.Client
+	remoteURL              *url.URL
+	allowedURLRegexPattern *regexp.Regexp
+	signer                 Signer
+	authenticator          Authenticator
+	reverseProxy           *httputil.ReverseProxy
 }
 
 // isValidTargetEndpoint checks if the request has a valid target depending
 // on the configuration of the proxy
 func (p *Proxy) isValidTargetEndpoint(r *http.Request) bool {
-	return r.URL.Path == BulkLogsEndpoint
+	return p.allowedURLRegexPattern.MatchString(r.URL.Path)
 
 }
 
 func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	if !p.isValidTargetEndpoint(r) {
-		unauthorized(w)
-		return
+	if p.config.AllowedPathRegex != "" {
+		if !p.isValidTargetEndpoint(r) {
+			unauthorized(w)
+			return
+		}
 	}
 
 	if p.config.InputValidation {
 		err := p.authenticator.AuthenticateRequest(r)
 		switch err.(type) {
+		case nil:
+			log.Trace("Authentication succesful")
 		case ErrIDNotWhitelisted:
 		case ErrSignatureInvalid:
 			msg := "not authorized to proxy request"
 			http.Error(w, msg, http.StatusUnauthorized)
+
+			log.
+				WithField("error", err).
+				WithField("proxy-id", r.Header.Get(ProxyIDHeader)).
+				Warning("AuthenticateRequest not authorized to proxy request")
```

Comment:
Why did you decide to add this log message?

Reply:
The trace was added to have something for the `nil` case, the warning is basically for us to know if someone tries to authenticate and fails for whatever reason, previously we wouldn't know :)